pkg/controllers/management/auth/project_cluster_handler.go

```diff
@@ -335,6 +341,32 @@ func (m *mgr) createProject(name string, cond condition.Cond, obj runtime.Object
 	})
 }
 
+// deleteSystemProject deletes the system project(s) for a cluster in preparation for deleting the cluster namespace.
+// Normally, the webhook prevents deleting the system project, so Rancher needs to use the sudo user to force it.
+// Otherwise, the deleted namespace will be stuck terminating because it cannot garbage collect the project.
+func (m *mgr) deleteSystemProject(obj runtime.Object, controller string) error {
```

Comment:
Two things:

1. Could this method operate not on `runtime.Object`?

2. Is the `controller` parameter needed? Do you plan to call this method elsewhere with different values? If not, then you could just use that constant in the code directly...